account/tests.py

```diff
@@ -0,0 +1,155 @@
+
+PROFESSIONAL_TYPE = UserType.Professional
+CLIENT_TYPE = UserType.Client
+PHONE_NUMBER = '1234567890'
+COUNTRY = 'testcountry'
+CITY = 'testcity'
+ADDRESS = 'testaddress'
+
+PROFESSION = Professions.Handyman
+DESCRIPTION = 'test description'
+
+BIRTHDAY = datetime.date(2000, 1, 1)
+
+
+@pytest.fixture
+def user():
+    user = User.objects.create_user(
+        username=USERNAME,
+        password=PASSWORD,
+        first_name=FIRST_NAME,
+        last_name=LAST_NAME,
+        email=EMAIL,
+        last_login=LAST_LOGIN
+    )
+    return user
+
+
+@pytest.fixture
+def profile(user):
+    profile = Profile.objects.create(
+        user_id=user,
+        user_type=PROFESSIONAL_TYPE,
+        phone_number=PHONE_NUMBER,
+        country=COUNTRY,
+        city=CITY,
+        address=ADDRESS
+    )
+    return profile
+
+
+@pytest.fixture
+def professional(profile):
+    professional_profile = profile
+    professional_profile.user_type = PROFESSIONAL_TYPE,
+    professional = Professional.objects.create(
+        profile_id=professional_profile,
+        profession=PROFESSION,
+        description=DESCRIPTION
+    )
+    return professional
+
+
+@pytest.fixture
+def client(profile):
+    client_profile = profile
+    client_profile.user_type = CLIENT_TYPE,
+    client = Client.objects.create(
+        profile_id=profile,
+        birthday=BIRTHDAY
+    )
+    return client
+
+
+@pytest.mark.django_db
+class TestProfileModel:
+    def test_new_profile(self, profile):
+        assert profile.user_type == PROFESSIONAL_TYPE
+        assert profile.phone_number == PHONE_NUMBER
+        assert profile.country == COUNTRY
+        assert profile.city == CITY
+        assert profile.address == ADDRESS
+
+    def test_get_profile(self, profile):
+        profile.user_id.save()
+        profile.save()
+        assert profile in Profile.objects.all()
+
+    def test_delete_profile(self, profile):
+        profile.user_id.save()
+        profile.save()
+        profile.delete()
+        assert profile not in Profile.objects.all()
+
+    def test_delete_user_deletes_profile(self, profile):
+        profile.user_id.save()
+        profile.save()
+        profile.user_id.delete()
+        assert profile not in Profile.objects.all()
+
+
+@pytest.mark.django_db
+class TestProfessionalModel:
+    def test_new_professional(self, professional):
+        assert professional.profession == PROFESSION
+        assert professional.description == DESCRIPTION
+
+    def test_get_professional(self, professional):
+        professional.profile_id.user_id.save()
+        professional.profile_id.save()
+        professional.save()
+        assert professional in Professional.objects.all()
+
+    def test_delete_professional(self, professional):
+        professional.profile_id.user_id.save()
+        professional.profile_id.save()
+        professional.save()
+        professional.delete()
+        assert professional not in Professional.objects.all()
```

Comment:
What happens to Client/Pro?

Reply:
The professional is deleted due to this test,
and I have a similar function for client (test_delete_client)